File: E/1287-element-appearing-more-than-25-in-sorted-array/main.go
package main

import "fmt"

func findSpecialInteger(arr []int) int {
	total := len(arr)
	cur := -1
	cnt := 0
	for i := 0; i < len(arr); i++ {
		if cur == arr[i] {
			cnt++
		} else {
			if cnt*4 > total {
				return arr[i-1]
			}
			cur = arr[i]
			cnt = 1
		}
	}
	if cnt*4 > total {
		return arr[len(arr)-1]
	}
	return -1
}

func main() {
	fmt.Println("a")
}

File: E/1287-element-appearing-more-than-25-in-sorted-array/main.rs
struct Solution {}

impl Solution {
  pub fn find_special_integer(arr: Vec<i32>) -> i32 {
    let step = arr.len() / 4;

    if arr[arr.partition_point(|&x| x < arr[step]) + step] == arr[step] {
      return arr[step];
    }
    if arr[arr.partition_point(|&x| x < arr[step * 2 + 1]) + step] == arr[step * 2 + 1] {
      return arr[step * 2 + 1];
    }
    arr[step * 3 + 2]
  }
}
